src/main/java/ch/njol/skript/ScriptLoader.java

```diff
@@ -540,234 +479,135 @@ public static CompletableFuture<ScriptInfo> loadScripts(List<Config> configs, Op
 		
 		return CompletableFuture.allOf(scriptInfoFutures.toArray(new CompletableFuture[0]))
 			.thenApply(unused -> {
-				SkriptEventHandler.registerBukkitEvents();
-				
-				// After we've loaded everything, refresh commands their names changed
-				if (syncCommands.get()) {
-					if (CommandReloader.syncCommands(Bukkit.getServer()))
-						Skript.debug("Commands synced to clients");
-					else
-						Skript.debug("Commands changed but not synced to clients (normal on 1.12 and older)");
-				} else {
-					Skript.debug("Commands unchanged, not syncing them to clients");
+				try {
+					openCloseable.open();
+
+					scripts.stream()
+						.flatMap(script -> { // Flatten each entry down to a stream of Config-Structure pairs
+							return script.getStructures().stream()
+								.map(structure -> new NonNullPair<>(script, structure));
+						})
+						.sorted(Comparator.comparing(pair -> pair.getSecond().getPriority()))
+						.forEach(pair -> {
+							Script script = pair.getFirst();
+							if (getParser().getCurrentScript() != script)
+								getParser().setCurrentScript(script);
```

Comment:
There can be multiple scripts being loaded at the same time with this method, so sometimes `getCurrentScript` needs updated. This check exists because I didn't want to call `setCurrentScript` for the same Script object. However, this is actually probably something that `setCurrentScript` should just check for.

Indeed, we do need to do some additional resets of `currentScript` too